audformat/core/utils.py

```diff
@@ -438,6 +438,64 @@ def intersect(
     return index
 
 
+def iter_by_file(
+        obj: typing.Union[
+            pd.Index,
+            pd.Series,
+            pd.DataFrame,
+        ],
+) -> typing.Iterator[
+    typing.Tuple[
+        str,
+        typing.Union[pd.Index, pd.Series, pd.DataFrame],
+    ],
+]:
+    r"""Iterate over object by file.
+
+    Each iteration returns a file and the according sub-object.
+
+    Args:
+        obj: object conform to
+            :ref:`table specifications <data-tables:Tables>`.
```

Comment:
```suggestion
            :ref:`table specifications <data-tables:Tables>`
```

We usually don't end with a dot here